File: monitoreo/src/main/java/models/entidades/comunidad/Comunidad.java
package models.entidades.comunidad;

import models.entidades.incidente.Incidente;
import models.entidades.incidente.Estado;
import models.entidades.persona.Miembro;
import models.entidades.servicios.servicio.PrestacionDeServicio;
import lombok.Getter;
import lombok.Setter;
import models.modulos.Notificador.Notificaciones.Notificacion.NuevoIncidente;
import models.modulos.Notificador.Notificador;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "comunidad")
public class Comunidad {

  @Id
  @Setter@Getter
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private int id;

  @Getter @Setter
  @Column(name = "nombre")
  private String nombre;

  @Getter
  @OneToMany(mappedBy = "comunidad", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, fetch = FetchType.LAZY)
  private List<Miembro> miembros;

  @Getter
  @OneToMany(mappedBy = "comunidad", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, fetch = FetchType.LAZY)
  private List<Incidente> incidentesReportados;

  @Getter @Setter
  @Column(name = "gradoDeConfianza")
  private Double gradoDeConfianza;

  @Getter
  @ManyToMany(fetch = FetchType.LAZY)
  private List<PrestacionDeServicio> serviciosParticularesObservados;

  @Getter @Setter
  @Column(name = "estaActiva")
  private Boolean estaActiva  = true;

  public Comunidad(String nombre) {
    this.nombre = nombre;
    this.miembros = new ArrayList<>();
    this.incidentesReportados = new ArrayList<>();
    this.serviciosParticularesObservados = new ArrayList<>();
  }

  public Comunidad(String nombre, Double gradoDeConfianza, List<PrestacionDeServicio> serviciosParticularesObservados){
    this.nombre = nombre;
    this.gradoDeConfianza = gradoDeConfianza;
    this.miembros = new ArrayList<>();
    this.incidentesReportados = new ArrayList<>();
    this.serviciosParticularesObservados = serviciosParticularesObservados;
  }

  public Comunidad(String nombre, Double gradoDeConfianza){
    this.nombre = nombre;
    this.gradoDeConfianza = gradoDeConfianza;
    this.miembros = new ArrayList<>();
    this.incidentesReportados = new ArrayList<>();
    this.serviciosParticularesObservados = new ArrayList<>();
  }

  public Comunidad(){
    this.miembros = new ArrayList<>();
    this.incidentesReportados = new ArrayList<>();
    this.serviciosParticularesObservados = new ArrayList<>();
  }

  public void agregarMiembro(Miembro miembro) {
    this.miembros.add(miembro);
    miembro.setComunidad(this);
  }

  public void removerMiembro(Miembro miembro) {
    this.miembros.remove(miembro);
  }

  public void agregarIncidenteReportado(Incidente incidente, Notificador notificador) {
    this.incidentesReportados.add(incidente);
    incidente.setComunidad(this);

    NuevoIncidente nuevoIncidente = new NuevoIncidente();
    for (Miembro miembro: this.miembros){
      notificador.notificarAMiembro(miembro, nuevoIncidente, incidente);
    }
  }

  public List<Incidente> incidentesSegunEstado(Estado estado){
    return incidentesReportados.stream().filter(incidente -> estado == incidente.getEstado()).toList();
  }

  public void agregarServicioObservado(PrestacionDeServicio prestacionDeServicio) {
  this.serviciosParticularesObservados.add(prestacionDeServicio);
    
  }

}

File: MonitoreoCNET/MonitoreoCNET/Models/Comunidades/Comunidad.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using MonitoreoCNET.Models.Incidentes;
using MonitoreoCNET.Models.Persona;
using MonitoreoCNET.Models.Servicios;


namespace MonitoreoCNET.Models.Comunidades
{
    [Table("comunidad")]
    public class Comunidad
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Column("nombre")]
        public string Nombre { get; set; }

        //[OneToMany(nameof(Miembro.Comunidad))]
        public virtual List<Miembro> Miembros { get; set; }

//[OneToMany(nameof(Incidente.Comunidad))]
        public virtual List<Incidente> IncidentesReportados { get; set; }

        [Column("gradoDeConfianza")]
        public double GradoDeConfianza { get; set; }

        //[ManyToMany]
        public virtual List<PrestacionDeServicio> ServiciosParticularesObservados { get; set; }

        public Comunidad(string nombre)
        {
            Nombre = nombre;
            Miembros = new List<Miembro>();
            IncidentesReportados = new List<Incidente>();
            ServiciosParticularesObservados = new List<PrestacionDeServicio>();
        }

        public Comunidad()
        {
            Miembros = new List<Miembro>();
            IncidentesReportados = new List<Incidente>();
            ServiciosParticularesObservados = new List<PrestacionDeServicio>();
        }

        public void AgregarMiembro(Miembro miembro)
        {
            Miembros.Add(miembro);
            miembro.Comunidad = this;
        }

        public void RemoverMiembro(Miembro miembro)
        {
            Miembros.Remove(miembro);
        }

        public void AgregarIncidenteReportado(Incidente incidente/* Notificador notificador*/)
        {
            IncidentesReportados.Add(incidente);
            incidente.Comunidad = this;
            /*
            Incidente.Incidente nuevoIncidente = new Incidente.Incidente();
            foreach (var miembro in Miembros)
            {
                notificador.NotificarAMiembro(miembro, nuevoIncidente, incidente);
            }+
            */
        }

        public List<Incidente> IncidentesSegunEstado(Estado estado)
        {
            return IncidentesReportados.FindAll(incidente => estado == incidente.Estado);
        }

        public void AgregarServicioObservado(PrestacionDeServicio prestacionDeServicio)
        {
            ServiciosParticularesObservados.Add(prestacionDeServicio);
        }
    }
}